core/server/master/src/main/java/alluxio/master/metastore/caching/CachingInodeStore.java

```diff
@@ -407,15 +409,15 @@ public EdgeCache(CacheConfiguration conf) {
      * @param option  the read options
      * @return the children
      */
-    public Map<String, Long> getChildIds(Long inodeId, ReadOption option) {
+    public SortedMap<String, Long> getChildIds(Long inodeId, ReadOption option) {
       if (mBackingStoreEmpty) {
-        return mIdToChildMap.getOrDefault(inodeId, Collections.emptyMap());
+        return mIdToChildMap.getOrDefault(inodeId, Collections.emptySortedMap());
       }
       // This implementation must be careful because edges can be asynchronously evicted from the
       // cache to the backing store. To account for this, we read from the cache before consulting
       // the backing store.
-      Map<String, Long> childIds =
-          new HashMap<>(mIdToChildMap.getOrDefault(inodeId, Collections.emptyMap()));
+      SortedMap<String, Long> childIds = new ConcurrentSkipListMap<>(mIdToChildMap.getOrDefault(
```

Comment:
Here are some (noisy) benchmarks using the JMH inode benchmarks.

In operations per second
Creating inodes:
- Skiplist: 194611
- Hashtable: 223577

Reading random Inodes (5 nested directories, 100 inodes per directory):
- Skiplist: 1085138
- Hashtable :1081249

Reading random Inodes (5 nested directories, 10000 inodes per directory):
- Skiplist: 952346
- Hashtable: 1141196

Listing directory (5 nested directories, 100 inodes per directory):
- Skiplist: 514775
- Hashtable: 424466

So it looks like writing is ~10% slower. With small directories, getting single file info is similar, but the skip list slows down as the directories get larger. And listing directories is faster with the skip list (probably because you don't have to traverse different buckets).

Reply:
Looks good to me, especially it's a relief to see <20% slowdown for a dir with 10k files. I think it's a good solution. Thanks for doing that :)